test/suite/test_error_info.py

```diff
@@ -26,15 +26,72 @@
 # ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
 # OTHER DEALINGS IN THE SOFTWARE.
 
+import errno
 import wiredtiger, wttest, time
 from wtdataset import SimpleDataSet
+from compact_util import compact_util
+
+import unittest
 
 # test_error_info.py
 #   Test that the placeholder get_last_error() session API returns placeholder error values.
-class test_error_info(wttest.WiredTigerTestCase):
+class test_error_info(compact_util):
+    table_name1 = 'table:test_error'
 
-    def test_error_info(self):
+    def check_error(self, error, sub_level_error, error_msg):
         err, sub_level_err, err_msg = self.session.get_last_error()
-        self.assertEqual(err, 0)
-        self.assertEqual(sub_level_err, wiredtiger.WT_NONE)
-        self.assertEqual(err_msg, "")
+        self.assertEqual(err, error)
+        self.assertEqual(sub_level_err, sub_level_error)
+        self.assertEqual(err_msg, error_msg)
+
+    def test_compaction_already_running(self):
+        # Enable the background compaction server.
+        self.turn_on_bg_compact()
+
+        # Attempt to reconfigure.
+        self.assertRaisesException(wiredtiger.WiredTigerError, lambda: self.session.compact(None, f'background=true,free_space_target=10MB'))
+
+        # Expect error code, sub-error code and error message to reflect compaction already running.
+        self.check_error(errno.EINVAL, wiredtiger.WT_BACKGROUND_COMPACT_ALREADY_RUNNING, "Cannot reconfigure background compaction while it's already running.")
+
+    def test_uncommitted_data(self):
+        # Create a simple table.
+        self.session.create(self.table_name1, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.table_name1)
+
+        # Start a transaction and insert a key and value.
+        self.session.begin_transaction()
+        cursor.set_key('key')
+        cursor.set_value('value')
+        self.assertEqual(cursor.update(), 0)
+        cursor.close()
+
+        # Attempt to drop the table without committing the transaction.
+        self.assertRaisesException(wiredtiger.WiredTigerError, lambda: self.session.drop(self.table_name1, None))
+
+        # Expect error code, sub-error code and error message to reflect uncommitted data.
+        self.check_error(errno.EBUSY, wiredtiger.WT_UNCOMMITTED_DATA, "the table has uncommitted data and can not be dropped yet")
+
+    def test_dirty_data(self):
+        # Create a simple table.
+        self.session.create(self.table_name1, 'key_format=S,value_format=S')
+        cursor = self.session.open_cursor(self.table_name1)
+
+        # Start a transaction, insert a key and value, and commit the transaction.
+        self.session.begin_transaction()
+        cursor.set_key('key')
+        cursor.set_value('value')
+        self.assertEqual(cursor.update(), 0)
+        self.assertEqual(self.session.commit_transaction(), 0)
+        cursor.close()
+
+        time.sleep(1)
```

Comment:
Please add the following comment now:

```suggestion
        # Give time for the oldest id to update.
        time.sleep(1)
```